docs/syntax-table-destructuring.md

```diff
@@ -0,0 +1,143 @@
+# Table Destructuring Syntax
+
+
+
+## Summary
+Introduce new syntax for unpacking values from tables.
+
+```lua
+local .{ .foo, one, ["@bar"] as bar } = thing
```

Comment:
Wanted to note that if we're going to have `.foo`, the outer `.` is actually superfluous. It's like the `://` in URIs, which only needed `:` and makes `//` superfluous. Once removed, this ends up being the same exact design as https://github.com/luau-lang/rfcs/pull/24.

Reply:
I guess with the exception of being able to perform assignments to locals/globals (not lvalues in general), but I've already said I am against that.